src/Projection/InMemoryEventStoreQueryFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Projection;
+
+use Prooph\EventStore\EventStore;
+
+final class InMemoryEventStoreQueryFactory implements QueryFactory
+{
+    /**
+     * @var EventStore
+     */
+    private $eventStore;
+
+    public function __construct(EventStore $eventStore)
+    {
+        $this->eventStore = $eventStore;
+    }
+
```

Comment:
@prolic Why not using a callable only? Is the setter really needed? /cc @codeliner 

```
public function __invoke(ProjectionOptions $options = null) {

}
```